### PR TITLE
Resolve hostname try catch fixes #6121

### LINKED ISSUE #6121: [Bug] Resolve-DbaNetworkName - some commands are missing credential and try catch.
Some commands are missing credential and try catch when using Resolve-DbaNetworkName.

To quote Shawns feedback on another command:-

"Need to provide the Credential parameter and variable. As well, need to wrap this in a try/catch block and add the -EnableException parameter so the catch block is properly called. This ensures error output is captured properly by the logging system, in the event the underlying function is hosed up by some other change/commit."

### PR DESCRIPTION
**Pull request by garethnewman**
fixes #6121

 - [X] Bug fix (non-breaking change, fixes #6121

Some commands are using Resolve-DbaNetworkName but not passing in credential or using try/catch.

### CODE REVIEW

File: functions/Disable-DbaForceNetworkEncryption.ps1 [hunk 1]
```diff
@@ -63,9 +63,13 @@ function Disable-DbaForceNetworkEncryption {
             Write-Message -Level VeryVerbose -Message "Processing $instance." -Target $instance
             $null = Test-ElevationRequirement -ComputerName $instance -Continue
 
-            Write-Message -Level Verbose -Message "Resolving hostname."
-            $resolved = $null
-            $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
+                $resolved = $null
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -ErrorAction Stop
```
Comment (wsmelton) [inline functions/Disable-DbaForceNetworkEncryption.ps1]: Please remove the ErrorAction and only use `-EnableException`

File: functions/Enable-DbaForceNetworkEncryption.ps1 [hunk 2]
```diff
@@ -63,9 +63,13 @@ function Enable-DbaForceNetworkEncryption {
             Write-Message -Level VeryVerbose -Message "Processing $instance." -Target $instance
             $null = Test-ElevationRequirement -ComputerName $instance -Continue
 
-            Write-Message -Level Verbose -Message "Resolving hostname."
-            $resolved = $null
-            $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
+                $resolved = $null
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -ErrorAction Stop
```
Comment (wsmelton) [inline functions/Enable-DbaForceNetworkEncryption.ps1]: Remove `-ErrorAction Stop` and replace with just `-EnableException`

File: functions/Get-DbaForceNetworkEncryption.ps1 [hunk 3]
```diff
@@ -53,9 +53,13 @@ function Get-DbaForceNetworkEncryption {
             Write-Message -Level VeryVerbose -Message "Processing $instance" -Target $instance
             $null = Test-ElevationRequirement -ComputerName $instance -Continue
 
-            Write-Message -Level Verbose -Message "Resolving hostname"
-            $resolved = $null
-            $resolved = Resolve-DbaNetworkName -ComputerName $instance
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
+                $resolved = $null
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -ErrorAction Stop
```
Comment (wsmelton) [inline functions/Get-DbaForceNetworkEncryption.ps1]: Remove `-ErrorAction Stop` and replace with just `-EnableException`

File: functions/Remove-DbaNetworkCertificate.ps1 [hunk 4]
```diff
@@ -66,9 +66,13 @@ function Remove-DbaNetworkCertificate {
             Write-Message -Level VeryVerbose -Message "Processing $instance" -Target $instance
             $null = Test-ElevationRequirement -ComputerName $instance -Continue
 
-            Write-Message -Level Verbose -Message "Resolving hostname"
-            $resolved = $null
-            $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
+                $resolved = $null
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -ErrorAction Stop
```
Comment (wsmelton) [inline functions/Remove-DbaNetworkCertificate.ps1]: Remove `-ErrorAction Stop` and replace with just `-EnableException`

File: functions/Set-DbaNetworkCertificate.ps1 [hunk 5]
```diff
@@ -92,9 +92,13 @@ function Set-DbaNetworkCertificate {
             Write-Message -Level VeryVerbose -Message "Processing $instance" -Target $instance
             $null = Test-ElevationRequirement -ComputerName $instance -Continue
 
-            Write-Message -Level Verbose -Message "Resolving hostname"
-            $resolved = $null
-            $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
+            try {
+                Write-Message -Level Verbose -Message "Resolving hostname."
+                $resolved = $null
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Credential $Credential -ErrorAction Stop
```
Comment (wsmelton) [inline functions/Set-DbaNetworkCertificate.ps1]: Remove `-ErrorAction Stop` and replace with just `-EnableException`

File: functions/Disable-DbaForceNetworkEncryption.ps1 [hunk 6]
```diff
@@ -63,9 +63,13 @@ function Disable-DbaForceNetworkEncryption {
...
+            } catch {
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```
Comment (wsmelton) [inline functions/Disable-DbaForceNetworkEncryption.ps1]: The `-Credential $Credential` should be included in this call.

File: functions/Enable-DbaForceNetworkEncryption.ps1 [hunk 7]
```diff
@@ -63,9 +63,13 @@ function Enable-DbaForceNetworkEncryption {
...
+            } catch {
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```
Comment (wsmelton) [inline functions/Enable-DbaForceNetworkEncryption.ps1]: The `-Credential $Credential` should be included in this call.

File: functions/Get-DbaForceNetworkEncryption.ps1 [hunk 8]
```diff
@@ -53,9 +53,13 @@ function Get-DbaForceNetworkEncryption {
...
+            } catch {
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```
Comment (wsmelton) [inline functions/Get-DbaForceNetworkEncryption.ps1]: The `-Credential $Credential` should be included in this call.

File: functions/Remove-DbaNetworkCertificate.ps1 [hunk 9]
```diff
@@ -66,9 +66,13 @@ function Remove-DbaNetworkCertificate {
...
+            } catch {
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```
Comment (wsmelton) [inline functions/Remove-DbaNetworkCertificate.ps1]: The `-Credential $Credential` should be included in this call.

File: functions/Set-DbaNetworkCertificate.ps1 [hunk 10]
```diff
@@ -92,9 +92,13 @@ function Set-DbaNetworkCertificate {
...
+            } catch {
+                $resolved = Resolve-DbaNetworkName -ComputerName $instance -Turbo
```
Comment (wsmelton) [inline functions/Set-DbaNetworkCertificate.ps1]: The `-Credential $Credential` should be included in this call.

**Comment by potatoqualitee** [comment]
awesome thank you both!

**Pull request merged**